app/src/main/java/com/example/musicapp/service/BoundService.kt

```diff
@@ -0,0 +1,213 @@
+package com.example.musicapp.service
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.media.MediaPlayer
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import androidx.core.net.toUri
+import com.example.musicapp.R
+import com.example.musicapp.data.model.Song
+import com.example.musicapp.receiver.Receiver
+import com.example.musicapp.view.*
+import com.example.mvpmusicapp.ui.ServiceInterface
+
+class BoundService : Service() {
+    private lateinit var media: MediaPlayer
+    private val binder = MyBinder()
+    private var notificationManager: NotificationManager? = null
+    var isPlaying = false
+    var songHolder = 0
+    private var prgCallback: ServiceInterface? = null
+    val songList = mutableListOf<Song>()
+
+    override fun onCreate() {
+        super.onCreate()
+        notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager?
+    }
+    inner class MyBinder : Binder() {
+        fun getBoundService(): BoundService = this@BoundService
+    }
+
+    override fun onBind(p0: Intent?): IBinder {
+        initReceiver()
+        return binder
+    }
+
+    fun setData(list: MutableList<Song>) {
```

Comment:
nếu list ko dùng mutable thì khai báo List<Song> thôi
```suggestion
    fun setData(list: List<Song>) {
```